runtime/hsa-runtime/core/driver/xdna/amd_xdna_driver.cpp

```diff
@@ -316,6 +317,16 @@ hsa_status_t XdnaDriver::InitDeviceHeap() {
   return HSA_STATUS_SUCCESS;
 }
 
+hsa_status_t XdnaDriver::GetHandleMappings(std::unordered_map<uint32_t, void*> &vmem_handle_mappings) {
```

Comment:
We could return a `const std::unordered_map<uint32_t, void*>&` and avoid the copy here. Do we ever expect the function to fail?